code/modules/events/radiation_storm.dm

```diff
@@ -42,8 +42,36 @@
 		lights()
 
 /datum/event/radiation_storm/proc/radiate()
+	var/radiation_level = rand(15, 30)
+	for(var/z in affecting_z)
+		SSradiation.z_radiate(locate(1, 1, z), radiation_level, 1)
+		
 	for(var/mob/living/C in living_mob_list)
-		C.apply_radiation_effects()
+		var/area/A = get_area(C)
+		if(!A)
+			continue
+		if(isNotStationLevel(A.z))
+			continue
+		if(A.flags & RAD_SHIELDED)
+			continue
+		if(istype(C,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = C
+			
+			if(H.is_diona())
+				var/damage = rand(15, 30)
+				H.adjustToxLoss(-damage)
+				if(prob(5))
+					damage = rand(20, 60)
+					H.adjustToxLoss(-damage)
+				to_chat(H, SPAN_NOTICE("You can feel flow of energy which makes you regenerate."))
+			
+			if(prob(5 * (1 - H.get_blocked_ratio(null, DAMAGE_RADIATION, damage_flags = DAMAGE_FLAG_DISPERSED, armor_pen = radiation_level))))
+				if (prob(75))
+					randmutb(H) // Applies bad mutation
+					domutcheck(H,null,MUTCHK_FORCED)
+				else
+					randmutg(H) // Applies good mutation
+					domutcheck(H,null,MUTCHK_FORCED)
```

Comment:
Thats some special snowflake code for a single event that should be generalized or removed.

-> The diona stuff is already handled in `/mob/living/carbon/proc/diona_handle_radiation(var/datum/dionastats/DS)`
-> The mutation stuff should be purged as its generally rather lrp (if we want more severe radiation effects then `/mob/living/carbon/human/handle_mutations_and_radiation()` should be used.

You could consider increasing the radiation strength to compensate for the lack of "additional action" in this proc.

Reply:
Made the above change to TRUE. Also nuked the snowflake code/mutation code, made the event output higher radiation levels, and increased the strength of radiation, especially at higher radiation levels. Extreme exposure is deadlier, though still treatable with proper medical care